### DISCUSSION TITLE
Please add option to pay for extra subdomains

### DISCUSSION THREAD

**Discussion by CristianKerr** [simple-login/app#1031] [Feature Requests]
I have 5 subdomains, but I would like to have more. I understand you can not hand over tons of subdomains for free, but you could add an option to pay monthy for extra subdomains. Thank you

**Comment by nguyenkims** [comment]
@CristianKerr I would suggest to buy a domain instead as it gives you more flexibility. You can find a .com domain which is less than $10/year and create an infinite amount of subdomains with it.

**Comment by CristianKerr** [comment]
@nguyenkims but my own domain will be publicly linked with my name, which makes it not anonymous.

**Reply by nguyenkims** [reply]
You can choose to hide your whois, most domain registrars support this option.
In this case if your domain doesn't contain your personal info and if you only use it on SimpleLogin (and not create a personal website with it), there's no way to know if it's a domain provided by SimpleLogin or it is your personal domain.

**Comment by vertigo220** [comment]
I've looked into getting my own domain before but have never been able to find anything I like that hasn't already been taken. Beyond that, many people, like myself, might want to use a service like this but not be willing or able to deal with getting their own domain, and that can be a real barrier. Being able to simply use an already available one is much quicker and easier, but when there are only a few options and, IMO, none of them are good for an email address, that hurts adoption. I just signed up for a trial account and, if there was a good subdomain option available, I'd be testing it out right now and hopefully switching over to this service, as it looks great. But I'm not going to use any of the available subdomains, and it's unlikely I'm going to figure out a personal domain anytime soon, which means transitioning over to this (and now is a great time for me to have done so) likely won't happen, unfortunately.

**Reply by Coder84619** [reply]
I personally use CloudFlare, which has registration and DNS. Pretty sure at some of the process to setup DNS records for a custom domain has external API access. So SimpleLogin could focus at least at the start one domain/DNS provider like Cloudflair and automate as much of that process as possible, and provide a simple how-to guide for the manual steps. Getting a domain with CF is easy, and setting up the DNS records is the "hardest" part and that is supported by APIs and could be automated.

**Reply by vertigo220** [reply]
Helping make the technical process easier would certainly help, but while I can't speak to the difficulty of that, since I've never even gotten that far, for me the difficult part that has always prevented me from getting my own domain is just coming up with a name. I've tried dozens, if not hundreds, of names, and they're all taken, all the good ones that is. At least if there were something like simplemail.com or smail.com (probably both taken, but just examples), that would provide much easier entry to people looking to use the service. The current choices are just very poor IMO.

**Reply by Coder84619** [reply]
Well good registrars will provide you long lists of domain name options based on parameters you can input. But I do agree that getting a name you like isn't always easy. But if it's just for email aliases with SimpleLogin, then IMHO the name doesn't matter much since the identity is best not tied obviously back to you.

**Reply by vertigo220** [reply]
The name still matters because it should be fairly short and easy to say and require the least amount of explaining when telling it to someone. I've found over the years it's best to have it as simple as possible, because people get confused over basic stuff like email addresses _all the time_ which leads to wasting time spelling it out for them and sometimes still having them screw it up. For example, with 8shield.net you'd always have to say "eight-shield dot net, with the number 8," and you'd probably run into people that don't know how to spell shield. Not to mention that, IMO, it just sounds strange having "shield" for an email domain. Most email services have "mail" in the name for a reason. With aleeas.com, not only does it sound fake, but you'd have to spell it out because people would assume it's "alias." As for slmail.me, it's one of the better ones, but you can't just say it as trying to do so would sound like you're slurring, and so you'd always have to spell it out. Plus people are used to .com and, to a lesser extent, .net and .org, but anything else is likely to trip a lot of people up. I know you wouldn't think so, but after having given out various email addresses for years, I have found people to just be really dumb with this stuff.

I've tried multiple registrars and haven't had luck with any of them in finding something good. I guess I'll try again, but I'll probably just end up giving up after a day or two like I always do. I've heard lots of good things about Hover, so that's the one I primarily focus on. I'm pretty wary of CloudFlare and would never use GoDaddy or Google. But as for the above comment that you can find a domain for <$10/yr, I'm sure it's _possible_, but it's highly unlikely, as the vast majority of ones I've seen are $30+, which further diminishes the value of this service over one that's easier and cheaper due to having better domain names.

**Reply by Coder84619** [reply]
Well Cloudflare has excellent prices, and I own like 10 domains and renewal for each one is like $10-$12/yr.

**Reply by Coder84619** [reply]
I just did a random search on Cloudflare for domains. Used random words like "dog" "coin" "mail" etc and it each time it came up with dozens of available suggestions with .com, .net and .org addresses for about $10/yr.

**Reply by vertigo220** [reply]
Except I don't want a meaningless, random word for my domain, and I'm fairly certain mail.com/net/org (or the others for that matter) aren't available, and the results are probably garbage like 123mailxyz.com. If I'm going to buy and setup a domain, I want it to have some meaning and be easy to remember and to give out, not some random string of characters. I'm trying namecheap's beast mode right now to try and figure something out, but after spending close to an hour going through setting it up, I have a whopping eight results, and there's one less than $10, two less than $20, and the rest mostly $30+. The reason the ones you're looking at are so cheap is because they're not very good. Also, from what I've read, you can't register a domain on Cloudflare and can only transfer one already purchased elsewhere. Not sure if that's changed in the past couple years.

The bottom line is, even if I find something, it would be far cheaper and easier if SimpleLogin simply had a couple better domains to choose from, even at a small premium. And they might make more money if they're paying $20-30/yr (and especially $10/yr) and charging $5/yr to each person that chooses to use them. And I realize that what makes a "good" domain for this is subjective and they could add several more and there would still be someone not happy with any of them, but I think they could satisfy a lot more users by carefully selecting a few more. And considering this service, which only forwards emails, is more expensive than my actual email service, which does a lot more, yet they offer _dozens_ of domains and subdomains to choose from for aliases, it's hard to see the value here, especially considering I'd have to pay _more_ for this than I am for my actual email _and_ pay more still for my own domain just to have a decent one. And I'd be paying 2.5x as much as I'm currently paying for 33mail _before_ paying for my own domain, whereas the 33mail domain (33mail.com) is far better IMO than any available from SimpleLogin.

I was interested in making the switch because I prefer open-source, but it will be enough work going through and changing my email everywhere; having to come up with a domain name and register it as well only to pay an extra ~$30-50/yr is making me really have second thoughts. SimpleLogin needs to be competitive, and IMO it's not. Most people aren't going to be willing to pay more, whether $18 more or still more for their own domain, for a similar service just because it's open-source. I mean, most people use Gmail. So regardless of the issue of finding a domain name, my original point, and that of the OP, that they should add more domain names to choose from stands, because there needs to be more value to justify the higher cost compared to alternatives.

**Reply by Coder84619** [reply]
Seriously dude I'm trying to help. As I told you Cloudflare is SELLING domains for $10/yr, and I did three random word searches and came up with dozens of very usable options. I mean if you can't spent $30/yr on something you like, then probably doing the whole alias thing is not your cup of tea. Don't tell me you can't find good options for $10/yr when I told you exactly how to do it. I'm done.

**Reply by vertigo220** [reply]
I get that you're trying to help, but my point is that just because you can find some domains for that price, doesn't mean any of them are ones I'd want. What's usable to you isn't necessarily usable to me. Again, I've tried multiple registrars multiple times over the years. That's why I'm making the point that it doesn't matter, and SimpleLogin should make things, well, simpler by adding some additional domains, especially considering what they're charging. I've been using aliases for longer than most, and I have no problem spending more if it's worth it, but just because I don't want to spend more than I should have to and already am, especially for little added benefit and a worse domain name, doesn't make it not my cup of tea. Again, I appreciate that you're trying to help, but the fact is none of this should even be an issue because they should have better domain choices. You're trying to help by offering a solution to one problem when I'm focusing on another problem altogether, I'm simply elaborating why more choices without having to go that route are preferable.

**Reply by Coder84619** [reply]
Personally I'd NEVER use an alias service with their built-in domains. I always, always use a custom domain I own. Why? Using their domain locks you into their service forever unless you change dozens/hundreds of aliases to another service/domain. For example, I had 350 aliases at a competitor alias service. They offer NO CSV export. All I had to do change my 350 aliases was flip my MX record to SImpleLogin. Took 2 minutes. With SimpleLogin catch-all, all 350 of my aliases instantly flipped over. 

An alias domain, for my use case, is 100% website registrations. I don't care what the heck the domain is, as long as it has a well known top level domain like .com, .net, .org. I have a separate vanity domain name that I use for two-way email with trusted people that need my real email. 

So personally I don't think SimpleLogin needs more domains. What I think would be great is to partner with a cheap quality registrar like Cloudflare that also does DNS, so that it's a one stop shop for a custom domain.

**Reply by vertigo220** [reply]
I agree with that, and it's the main reason I've wanted to get my own domain for years, so it would be more portable. While I mostly use aliases for websites, I also have to give my email out over the phone a fair bit, which is why I prefer something easy to say and have the person understand, sometimes I have to type it on a touchscreen, which makes shorter better (not to mention shorter is better in general anyways), and, the biggest hurdle for me, is I also need to be able to use it professionally and semi-professionally, so I want it to fit the part. I already have always had a hard time coming up with names, so combined with the fact basically all the good ones are taken it makes it very hard for me to find something that will work well. I'd happily pay $50/yr for a really good domain name that works well for me, but unfortunately I haven't found any even at that price. So while having my own would be far superior, having a decent one that will work well enough that I don't have to think about and spend hours trying to figure out as an included option is the next-best thing. Maybe I just need to get someone more creative in this area and give them my criteria and see what they can come up with.

But again, even if I found the perfect domain name for $5/yr, it wouldn't change the other issue, which is that 33mail and AnonAddy are both $12/yr for similar features whereas this is $30/yr, so regardless of the domain issue, there's also an apparent value issue, and adding more domains would at least help with that a bit, especially if they can do so for the cost of one or two subs.

> I don't care what the heck the domain is, as long as it has a well known top level domain like .com, .net, .org.

In trying to find something decent, I always check other TLDs as well, though obviously some are automatic no-gos and some are more iffy than others. Have you found, or are you aware of, issues using emails with lesser-known TLDs? What I definitely don't want to do is buy a domain and get it set up and start switching everything over only to find websites don't like it.

**Reply by Coder84619** [reply]
About 1% of my 350 alias sites did NOT like my "uncommon" 4 letter domain TLD. So I recently added a second domain to SimpleLogin that has is a well known .com, .org, .net extension.

**Reply by vertigo220** [reply]
Well that's not too bad. And since it's only going to be websites, yeah, those can be anything. So at least that means I should be ok with a less common one. My bigger issue with them, as mentioned before, is that they're more likely to confuse people. People already have enough difficulty with "yourbusinessname @ alias.33maildotcom."

**Reply by JtwoA** [reply]
I've never seen someone combat help so hard in my life lol. I'm not giving you a hard time, @vertigo220, but you're sorta talking in circles.  You want an easy to communicate, personalized to "grab you", that is anonymous.  If it's that much "you" then it's no anonymous, just pointing that out.  Pick a word, buy a domain.  It doesn't NEED to make "sense".

I went a different path:  I used subdomains of my personal domain.  **e.** and **email.** work quite week and are easy to communicate.  Of course, I'm not trying to hide - I'm using SL for silo'ing every site I create logins for.  I want to know who is selling my address AND I want the ability to shut them off as necessary.  

I also want to use a personally owned domain for a reason someone mentioned above:  if I leave this service my aliases still work because I can recreate them at wherever my new service lands me.

**Comment by vjayer** [comment]
I'd be willing to pay too.

FYI anonaddy offers 10 of these sub-domains on their premium plan, called "usernames" (e.g. user.anonaddy.com) But they are limited in other ways compared to SimpleLogin
But I also second the suggestion https://github.com/simple-login/app/discussions/1031#discussioncomment-4607724
> What I think would be great is to partner with a cheap quality registrar like Cloudflare that also does DNS, so that it's a one stop shop for a custom domain

The primary reason why I want to use custom SL subdomains other than for organizing emails/aliases, is to work around the limitation on the standard SL domains of adding random dot extension to custom aliases even if our alias would be unique
I posted a feature request regarding that here: https://github.com/simple-login/app/discussions/1746
